.github/workflows/tag_release.yml

```diff
@@ -0,0 +1,74 @@
+name: Create Release and Attach Files
+
+on:
+  pull_request:
+    types:
+      - closed
+    branches:
+      - main
+
+jobs:
+  create_release:
+    if: ${{ github.event.pull_request.merged == true }}
```

Comment:
Something seems off about this unless I am misunderstanding.

I manually run the update_files workflow. That updates files, pushes to `dev` and opens a PR to `main`
I merge the PR to `main`
That triggers the `tag_release` workflow that creates the release based off the version in the CHANGELOG

When does the `release` workflow come into play?

I think you could replicate the Empire flow by checking that the PR being merged is a release PR https://github.com/BC-SECURITY/Empire/blob/main/.github/workflows/release-private-tag.yml#L11

Reply:
Almost, here is the flow:
1. You run "update_files", which creates a branch with the new .py files. It will also open a PR to the dev branch.
2. You run "release", which will update the changelog and version number. Then it will open a PR from dev to Main.
3. The "tag_release" gets triggered automatically when a PR is closed to main which creates a new release with the notes and files.